java-daemon-endpoints/src/main/java/net/cryptic_game/backend/endpoints/network/NetworkMemberEndpoints.java

```diff
@@ -1,10 +1,130 @@
 package net.cryptic_game.backend.endpoints.network;
 
-import net.cryptic_game.backend.base.api.endpoint.ApiEndpointCollection;
+import com.google.gson.JsonObject;
+import net.cryptic_game.backend.base.api.endpoint.*;
+import net.cryptic_game.backend.data.device.Device;
+import net.cryptic_game.backend.data.network.Network;
+import net.cryptic_game.backend.data.network.NetworkInvitation;
+import net.cryptic_game.backend.data.network.NetworkMember;
+import net.cryptic_game.backend.data.user.User;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
 
 public class NetworkMemberEndpoints extends ApiEndpointCollection {
 
     public NetworkMemberEndpoints() {
-        super("network/member");
+        super("network_member");
+    }
+
+    @ApiEndpoint("member")
+    public ApiResponse member(@ApiParameter("user_id") final UUID userId,
+                              @ApiParameter("device") final UUID deviceId) {
+        final User user = User.getById(userId);
+        final Device device = Device.getById(deviceId);
+
+        if (device == null || !device.hasUserAccess(user)) return new ApiResponse(ApiResponseType.FORBIDDEN, "DEVICE_ACCESS_DENIED");
+        if (!device.hasUserAccess(user))
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "DEVICE_ACCESS_DENIED");
+
+        if (!device.isPoweredOn())
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "DEVICE_NOT_ONLINE");
+
+        List<NetworkMember> memberships = NetworkMember.getMembershipsOfDevice(device);
+
+        return new ApiResponse(ApiResponseType.OK, memberships);
+    }
+
+    @ApiEndpoint("request")
+    public Object request(@ApiParameter("user") final UUID userId,
+                          @ApiParameter("device") final UUID deviceId,
+                          @ApiParameter("id") final UUID id) {
+        final User user = User.getById(userId);
+        final Device device = Device.getById(deviceId);
+        Network network = Network.getById(id);
+
+        if (device == null || !device.hasUserAccess(user)) {
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "DEVICE_ACCESS_DENIED");
+        }
+
+        if (!device.isPoweredOn()) {
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "DEVICE_NOT_ONLINE");
+        }
+
+        if (network == null) {
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "NETWORK_NOT_FOUND");
+        }
+
+        if (network.getOwner().equals(device)) {
+            return new ApiResponse(ApiResponseType.ALREADY_EXISTS, "ALREADY_MEMBER_OF_NETWORK");
+        }
+
+        if (NetworkMember.getMember(network, device) != null) {
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "ALREADY_MEMBER_OF_NETWORK");
+        }
+
+        if (NetworkInvitation.getInvitation(network, device) != null) {
+            return new ApiResponse(ApiResponseType.ALREADY_EXISTS, "INVITATION_ALREADY_EXISTS");
+        }
```

Comment:
Check if the existing NetworkInvitation is an invitation (and not a request). In case of the existing NetworkInvitation being an invitation accept it, otherwise return `REQUEST_ALREADY_EXISTS`.